### PR TITLE
Give pieces to players on initialisation

### PR DESCRIPTION
**Pull request by swcraig**
Upon instantiation a `Player` will receive their `pieces` as a hash containing an array of court pieces and an array of pawns.  Having players "come with" their pieces will make the future `Board` setup cleaner.

### CODE REVIEW

File: lib/player.rb [hunk 1]
```diff
@@ -1,8 +1,14 @@
+require_relative 'piece_set'
+
 # Represents a chess player
 class Player
+  include PieceSet
```
Comment (brendandeere) [inline lib/player.rb]: Why are you including this module here?

File: spec/player_spec.rb [hunk 2]
```diff
@@ -8,4 +8,18 @@
       expect(subject).to be_a Player
     end
   end
+
+  describe '#pieces' do
```
Comment (brendandeere) [inline spec/player_spec.rb]: Why are you testing an attribute reader? is that code you wrote? Can this behaviour be tested somewhere else? If this reader does need to be tested, thats an indication to me that something weird is going on here.

File: spec/player_spec.rb [hunk 3]
```diff
@@ -8,4 +8,18 @@
...
+    subject { Player.new('Siegbert').pieces }
+    it 'includes a row of pawns' do
+      pawns = subject[:pawns]
+      expect(pawns).to all be_a Pawn
+    end
+
+    it 'includes a row of court_pieces' do
```
Comment (brendandeere) [inline spec/player_spec.rb]: `:aggregate_failures` 

Write two expectations. one that tests that all of objects are pieces, another that tests that they are all not pawns.

**Comment by swcraig** [comment]
@brendandeere

**Pull request merged**